server/controllers/CommentController.js

```diff
@@ -0,0 +1,228 @@
+   * @param {object} res the response object
+   * @returns {object} an object containing an array of all comments.
+   */
+  static getComments(req, res) {
+    Comment.findAll({
+      include: [
+        { model: User, as: 'commenter' },
+      ]
+    }).then((comments) => {
+      const destructuredComments = comments.map(comment => Object.assign(
+        {},
+        {
+          id: comment.id,
+          parentId: comment.parentId,
+          createdAt: new Date(comment.createdAt).toLocaleString('en-GB', { hour12: true }),
+          updatedAt: new Date(comment.updatedAt).toLocaleString('en-GB', { hour12: true }),
+          body: comment.body,
+          author: {
+            username: comment.commenter.username,
+            bio: comment.commenter.bio,
+            image: comment.commenter.image,
+          },
+        }
+      ));
+      const result = CommentController.nestComments(destructuredComments);
+      res.status(200).json({
+        status: 'success',
+        comments: result,
+      });
+    }).catch(() => {
+      res.status(400).json({
+        status: 'fail',
+        message: 'Unable to get comments',
+      });
+    });
+  }
+
+  /**
+   * Get a comment by its id
+   * @param {object} req the request object
+   * @param {object} res the response object
+   * @returns {object} an object containing an array of all comments.
+   */
+  static getComment(req, res) {
+    Comment.findById((req.params.id), {
+      include: [
+        { model: User, as: 'commenter' },
+      ]
+    }).then(comment => res.status(200).json({
+      status: 'success',
+      comment: {
+        id: comment.id,
+        parentId: comment.parentId,
+        createdAt: new Date(comment.createdAt).toLocaleString('en-GB', { hour12: true }),
+        updatedAt: new Date(comment.updatedAt).toLocaleString('en-GB', { hour12: true }),
+        body: comment.body,
+        author: {
+          username: comment.commenter.username,
+          bio: comment.commenter.bio,
+          image: comment.commenter.image,
+        },
+      }
+    })).catch(() => {
+      res.status(400).json({
+        status: 'fail',
+        message: 'Unable to get comment.',
+      });
+    });
+  }
+
+  /**
+   * Update a comment
+   * @param {object} req the request object
+   * @param {object} res the response object
+   * @returns {object} the updated comment.
+   */
+  static updateComment(req, res) {
+    Comment.findById(parseInt(req.params.id, 10))
+      .then((comment) => {
+        if (!comment) {
+          return res.status(404).json({
+            status: 'fail',
+            message: 'No comment found, please check the id supplied',
+          });
+        }
+        Comment.update({
+          body: req.body.body,
+        }, {
+          returning: true,
+          where: { id: comment.id },
+        })
+          .then(([, [updatedComment]]) => {
+            res.status(200).json({
+              status: 'success',
+              data: {
```

Comment:
@madeofhuman Nice work! I must implore your creativity around the approach you took regarding this task, however based on the reference on the readme, the response returned should be "comment" and not "data".

Reply:
Nice catch! I'll update that. Thank you.